### PR TITLE
Draw Timetable

### LINKED ISSUE #106: Draw out timetable in the output

### PR DESCRIPTION
**Pull request by daniellimws**
Now we can draw out the timetable like this:

image

The task list will be printed at the bottom for the user's reference because some task names will be too long to show in the timetable.

`Timetable` draws the timetable based on the filtered task list, and a starting hour and an ending hour, which are constant for now (starting from 0800 to 2200) but later we probably need to allow the user to set this. Now, we can only draw the timetable for the week. The tasks will still be filtered based on the forecast input, but for now we only draw the timetable for the week (if the user filters `TODAY` then the timetable for the whole week is drawn out but only the tasks for today is shown). We can work on drawing the timetable based on the forecast in another PR.

Many test files were modified because in order for the timetable drawing to work, we need to set the `startTime` to be a `LocalTime` and duration to be an `int`. This means that from now on, our input for `startTime` must be in the `hhmm` format and `duration` must be an integer. We will need to catch exceptions for these but I think this PR is quite big already so I think later we open an issue for this?

I haven't added tests for this yet. I will do that tomorrow.

### CODE REVIEW

File: src/main/java/athena/timetable/Timetable.java
Comment (Kafcis) [inline src/main/java/athena/timetable/Timetable.java]: need more parser code to change this later

File: src/main/java/athena/timetable/Timetable.java
Comment (Kafcis) [inline src/main/java/athena/timetable/Timetable.java]: maybe can use the string "+-------"?

File: src/main/java/athena/timetable/Timetable.java
Comment (Kafcis) [inline src/main/java/athena/timetable/Timetable.java]: same thing as line 104,  maybe can use only 1 string

File: src/main/java/athena/timetable/Timetable.java
Comment (Kafcis) [inline src/main/java/athena/timetable/Timetable.java:258]: not sure if this is going to be flagged in the checkstyle

File: src/main/java/athena/timetable/Timetable.java
Comment (Kafcis) [inline src/main/java/athena/timetable/Timetable.java:311]: Something similar is in 115, could consider making it an attribute or a method call from the Task/Time class

**Review by Kafcis — COMMENTED** [review]
Nice changes, there are a few checkstyle issues but you'll probably fix them when you add the tests so I'm not going to comment on it

File: src/main/java/athena/timetable/Timetable.java
Comment (daniellimws) [reply src/main/java/athena/timetable/Timetable.java]: Oh I was planning to refactor `-------` into a constant. Will do that tomorrow.

File: src/main/java/athena/timetable/Timetable.java
Comment (daniellimws) [reply src/main/java/athena/timetable/Timetable.java:311]: It turns out that the one in 115 was looking for the first date that matches the day, e.g. given Wednesday return 21 Oct 2020. So it's different from this. But yea I agree it would be better for this to be somewhere else, so that it can also be commonly used by the date related tests.

File: src/main/java/athena/timetable/Timetable.java
Comment (daniellimws) [reply src/main/java/athena/timetable/Timetable.java:258]: Nope since the checks passed

File: src/main/java/athena/timetable/Timetable.java
Comment (alstontham) [inline src/main/java/athena/timetable/Timetable.java:146]: Do you want to shift the methods that print out the Timetable to Ui?

**Review by alstontham — COMMENTED** [review]
Timetable looks nice, just thinking if you want to shift the printing of it to Ui instead? Also would the formatting of the Timetable mess up if, for example two tasks overlap?

File: src/main/java/athena/timetable/Timetable.java
Comment (daniellimws) [reply src/main/java/athena/timetable/Timetable.java:146]: At first I was thinking of putting it in `Ui`, and then I feel that eventually we would need to refactor them to somewhere more focussed. Then I realized the somewhere is `Timetable`, since it contains all the related attributes needed to draw the timetable. What do you think?

**Comment by daniellimws** [comment]
Good catch. Currently if there are overlapping tasks, the second task will not show up in the timetable. I think we should leave this to the `TaskList` and `TimeAllocator` to ensure this doesn't happen? Because I don't think we should allow this to happen at all.

**Comment by daniellimws** [comment]
Btw for now, if the task duration exceeds the sleep time, it will be cut off at the sleep time. Perhaps we should push back the sleep time when this happens? We can open an issue for this if so.

File: src/main/java/athena/timetable/Timetable.java
Comment (Kafcis) [reply src/main/java/athena/timetable/Timetable.java:146]: I think that keeping these this here is better, it keeps calling bits and pieces all the time. If we do eventually do refactor it, the class needs to put the entire timetable in one shot

File: src/main/java/athena/timetable/Timetable.java
Comment (Kafcis) [inline src/main/java/athena/timetable/Timetable.java:232]: we can just throw an error that says that the tasknumber is conflict with another, I'll handle this in the allocator PR

File: src/main/java/athena/timetable/Timetable.java
Comment (Kafcis) [reply src/main/java/athena/timetable/Timetable.java:258]: awesome

**Review by Kafcis — APPROVED** [review]

File: src/main/java/athena/timetable/Timetable.java
Comment (daniellimws) [reply src/main/java/athena/timetable/Timetable.java:232]: Ok will leave this to you

**Review by amanda-chua — APPROVED** [review]
Timetable looks good! In this case, list without filters will show the tasks for the week only?

**Comment by daniellimws** [comment]
Ya it shows for the week for now. After this is merged I will let it be based on the `ForecastFilter` (#129).

File: src/main/java/athena/Parser.java
Comment (wish2023) [inline src/main/java/athena/Parser.java:176]: Remember to update this new default value in user guide

File: src/main/java/athena/Storage.java
Comment (wish2023) [inline src/main/java/athena/Storage.java]: Seems like you have a `getDurationString()` it may be easier to use that?

**Review by wish2023 — APPROVED** [review]
Looks good! Super minor comment about a method suggested

File: src/main/java/athena/Parser.java
Comment (daniellimws) [reply src/main/java/athena/Parser.java:176]: Oh thanks I forgot about that.

**Pull request merged**
